pkg/fetcher/fetcher.go

```diff
@@ -414,6 +428,72 @@ func (u *URLFetcher) IsNonretryableClientError() bool {
 		s != http.StatusLocked && s != http.StatusTooManyRequests
 }
 
+// buildRegistryErrMsg builds error message for unexpected http code (nonretryable client errors and all other errors)
+// and extracts message details from response body stream if there is one (#5951).
+func (u *URLFetcher) buildRegistryErrMsg(url *url.URL, respBody io.ReadCloser) string {
+	errMsg := fmt.Sprintf("Unexpected http code: %d (%s), URL: %s", u.StatusCode, http.StatusText(u.StatusCode), url)
+
+	errDetail, err := extractErrResponseMessage(respBody)
+	if err != nil {
+		return errMsg
+	}
+
+	errMsg += fmt.Sprintf(", Message: %s", errDetail)
+	return errMsg
+}
+
+// malformedJsonErrFormat is the error format for malformed json response body
+// used in function extractErrResponseMessage
+var malformedJsonErrFormat = fmt.Errorf("error response json has unconventional format")
```

Comment:
The idiom for declaring error variables is to prefix them with `err`. So `errJsonFormat` should be better here.